Fixed #35731 -- Extend documentation about db_default and DatabaseDefault. (#18682)
django/django: docs/ref/models/fields.txt

set at the database level and will be used when inserting rows outside of the
ORM or when adding a new field in a migration.

> .. note::
> 
>     ``DatabaseDefault()`` class is an internal class which provides a
>     mechanism to allow db_default to store values at database level.

Reviewer: I don't feel like this covered the confusion which was mentioned in the ticket, maybe something like...
```suggestion
If a field has a ``db_default`` without a :attr:`Field.default` set, unsaved
model instances which have not assigned a value to the field will return a
``DatabaseDefault``` object when accessing the field value. The value is
calculated when the model instance is saved to the database.
```

Author: Okay makes sense. Let me add both of the commits and go through the texts again.

Author: @sarahboyce Should we add thi

> I don't feel like this covered the confusion which was mentioned in the ticket, maybe something like...
> 
> Although the sentence is horribly long 🤔 but hopefully you see where I'm going with it

Should we put this message inside a note? What do you think?

Reviewer: ```suggestion
.. admonition:: The field value is a ``DatabaseDefault`` object prior to saving

    Unsaved model instances that have not set a value to a field with a
    ``db_default`` (without a ``default`` set) will return a ``DatabaseDefault``
    object when accessing the field value.

    .. code-block:: shell

        >>> my_model_instance = MyModel()
        >>> my_model_instance.created
        <django.db.models.expressions.DatabaseDefault object at 0x7a5d29cad610>

    The value is calculated when the model instance is saved to the database.
```
Maybe

Author: Okay i exactly had this in mind, but thought would become too lengthy. But here being more descriptive will help. Making the changes in the commit.

Reviewer: Docs are hard - we might iterate a bit before we're happy :+1:

Author: Totally! I get it! But in the latest commit i don't understand why the "Docs / blacken-docs (pull_request) " is failing.

set at the database level and will be used when inserting rows outside of the
ORM or when adding a new field in a migration.

If a field is configured with db_default but lacks a default, accessing the field
on an unsaved model instance, where no value has been explicitly assigned,
will return a DatabaseDefault object::

    from django.db import models
    from django.db.models.functions import Now

> 

Reviewer: ```suggestion

```
(As a guess) but running blacken-docs should sort it

Author: @sarahboyce It passed the test now, But does the changes seem appropriate now?

set at the database level and will be used when inserting rows outside of the
ORM or when adding a new field in a migration.

> If a field is configured with db_default but lacks a default, accessing the field
> on an unsaved model instance, where no value has been explicitly assigned,
> will return a DatabaseDefault object::
> 
>     from django.db import models
>     from django.db.models.functions import Now
> 
> 
>     class MyModel(models.Model):
>         # No default provided at the Django (Python) level, only db_default
>         created = models.DateTimeField(db_default=Now())
> 
> .. code-block:: shell
> 
>     >>> instance = MyModel()
>     >>> print(instance.created)
>     <django.db.models.expressions.DatabaseDefault object at 0x743257147ca0>
> 
> The actual value for the field is determined by the database when the model
> instance is saved.

Reviewer: Been playing around with the sentence structure and I think the following is clear
```suggestion
If a field has a ``db_default`` without a :attr:`~Field.default` set and no
value is assigned, a ``DatabaseDefault`` object is returned as the field value
on unsaved model instances. The actual value for the field is determined by the
database when the model instance is saved.
```
If this is clear enough, then we won't need the example

Author: I mean it is clear enough, So should we get rid of the  examples above? As i feel it solves the mentioned issue in the ticket and can also be understood without using a very detailed example...

Reviewer: I think so, yes
